diff --git a/libavutil/dict.c b/libavutil/dict.c
index 985208889b2..4a8ae0fea08 100644
--- a/libavutil/dict.c
+++ b/libavutil/dict.c
@@ -157,36 +157,35 @@ int av_dict_set_intptr(AVDictionary **pm, const char *key, uintptr_t value,
                 int flags)
 {
     char valuestr[22];
-    snprintf(valuestr, sizeof(valuestr), "%p", value);
+    snprintf(valuestr, sizeof(valuestr), "0x%"PRIxPTR, value);
     flags &= ~AV_DICT_DONT_STRDUP_VAL;
     return av_dict_set(pm, key, valuestr, flags);
 }
 
 uintptr_t av_dict_get_intptr(const AVDictionary *m, const char* key) {
-    uintptr_t ptr = NULL;
     AVDictionaryEntry *t = NULL;
     if ((t = av_dict_get(m, key, NULL, 0))) {
       return av_dict_strtoptr(t->value);
     }
-    return NULL;
+    return 0;
 }
 
 uintptr_t av_dict_strtoptr(char * value) {
-   uintptr_t ptr = NULL;
+   uintptr_t ptr = 0;
    char *next = NULL;
    if(!value || value[0] !='0' || (value[1]|0x20)!='x') {
-       return NULL;
+       return 0;
    }
    ptr = strtoull(value, &next, 16);
    if (next == value) {
-       return NULL;
+       return 0;
    }
    return ptr;
 }
 
 char * av_dict_ptrtostr(uintptr_t value) {
     char valuestr[22] = {0};
-    snprintf(valuestr, sizeof(valuestr), "%p", value);
+    snprintf(valuestr, sizeof(valuestr), "0x%"PRIxPTR, value);
     return av_strdup(valuestr);
 }

